src/coding-conventions.md

```diff
@@ -1,10 +1,83 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Coding conventions](#coding-conventions)
+  - [Linters/formatters](#lintersformatters)
+  - [TSDoc comments](#tsdoc-comments)
+  - [Lifecycle management](#lifecycle-management)
+  - [Component styles for different component states/variants](#component-styles-for-different-component-statesvariants)
+  - [Customizing components](#customizing-components)
+    - [Defining (default) component options](#defining-default-component-options)
+    - [Component variants with different options](#component-variants-with-different-options)
+      - [Areas to make them configurable as component options](#areas-to-make-them-configurable-as-component-options)
+      - [Areas where component optinos are _not_ applied](#areas-where-component-optinos-are-_not_-applied)
+  - [Polymorphism with static properties](#polymorphism-with-static-properties)
+  - [Custom events](#custom-events)
+  - [Custom element registration](#custom-element-registration)
+  - [Propagating misc attributes from shadow host to an element in shadow DOM](#propagating-misc-attributes-from-shadow-host-to-an-element-in-shadow-dom)
+    <!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
 # Coding conventions
 
-## Creating shadow root
+## Linters/formatters
+
+`carbon-custom-elements` uses ESLint with `typescript-eslint` for linting, and Prettier for code formatting.
+Most of ESLint configurations are same as ones in `carbon-components`.
+
+## TSDoc comments
+
+In addition to using TypeScript, we try to leverage editors' code assistance feature as much as possible.
+
+For that purpose, we add TSDoc comments to the following:
+
+- All classes
+- All properties/methods (including private properties), only exception here is one being overriden
+- All type definitions (e.g. `interface`, `enum`)
+
+## Lifecycle management
+
+To avoid memory leaks and zombie event listeners, we ensure the event listeners on custom elements themselves (hosts) and ones on `document`, etc. are released when they get out of render tree.
+
+For that purpose, `carbon-custom-elements` uses [`on()` API from `carbon-components`](https://github.com/carbon-design-system/carbon/blob/v10.2.0/src/globals/js/misc/on.js), a thin sugar layer of `.addEventListener()`.
```

Comment:
For host listeners it would be great if we could add something similar to Angulars [`HostListener`](https://angular.io/api/core/HostListener) ... automatically bind and clean up listeners as a decorator on a method.

Under the hood we can use the `cabon-components` `on()` to keep things simple

Reply:
Interesting point - I think this has perfect sense. Let me look into this.